src/license/LicenseInfoService.ts

```diff
@@ -0,0 +1,90 @@
+import { LGPLLicense } from './LGPLLicenseInfos';
+import { LibraryLicenseInfo, LicenseInfo } from './LicenseInfos';
+import { CplaceLicense } from './CplaceLicense';
+import * as fs from 'fs';
+import * as path from 'node:path';
+import { OtherLicense } from './OtherLicense';
+
+export const LIBRARY_LICENSE_INFOS_NAME = 'libraryLicenseInfos.json';
+
+function handleLibraryLicenceInfo(
+    libraryLicenseInfo: LibraryLicenseInfo,
+    pathToAssetsFolder: string,
+    licenseInfos: LicenseInfo[]
+) {
+    if (libraryLicenseInfo.licenseFile) {
+        const pathToLicenseFile = path.join(
+            pathToAssetsFolder,
+            libraryLicenseInfo.licenseFile
+        );
+        if (fs.existsSync(pathToLicenseFile)) {
+            libraryLicenseInfo.licenseText = fs
+                .readFileSync(pathToLicenseFile)
+                .toString()
+                .replace(/(\r\n|\n|\r)/gm, '');
+        }
+    } else {
+        console.log(
+            `Expected License File for ${libraryLicenseInfo.product} ${libraryLicenseInfo.component} does not exist ${libraryLicenseInfo.licenseFile}`
+        );
+    }
+
+    if (
+        libraryLicenseInfo.license
+            .toLowerCase()
+            .includes('commercial license') &&
+        libraryLicenseInfo.license
+            .toLowerCase()
+            .includes('collaboration factory')
+    ) {
+        licenseInfos.push(new CplaceLicense(libraryLicenseInfo));
+    } else if (isLgpl(libraryLicenseInfo)) {
+        licenseInfos.push(new LGPLLicense(libraryLicenseInfo));
+    } else {
+        licenseInfos.push(new OtherLicense(libraryLicenseInfo));
+    }
+}
+
+export function isLgpl(libraryLicenseInfo: LibraryLicenseInfo): boolean {
+    let includesLgpl = false;
+    if (libraryLicenseInfo.license.toLowerCase().includes('lgpl')) {
+        includesLgpl = true;
+    }
+    if (
+        libraryLicenseInfo.additionalLicenses &&
+        libraryLicenseInfo.additionalLicenses.length > 0
+    ) {
+        libraryLicenseInfo.additionalLicenses.forEach(
+            (additionalLicense: string) => {
+                if (additionalLicense.toLowerCase().includes('lgpl')) {
+                    includesLgpl = true;
+                }
+            }
+        );
+    }
+    return includesLgpl;
+}
+
+export function createLibraryLicenseInfos(pathToAssetsFolder: string) {
+    let libraryLicenseInfos: LibraryLicenseInfo[] = JSON.parse(
+        fs
+            .readFileSync(
+                path.join(pathToAssetsFolder, LIBRARY_LICENSE_INFOS_NAME)
+            )
+            .toString()
+    );
+
+    let licenseInfos: LicenseInfo[] = [];
+    libraryLicenseInfos.forEach((libraryLicenseInfo: LibraryLicenseInfo) => {
```

Comment:
Currently all license types are processed and added to compressed.js. We need to process and add only the types described in the ticket. We have to filter licenses by type.

Maybe having the whitelist in some service or common class like:

`public static LICENSE_TYPE_WHITELIST_PATTERNS = ['LGPL', 'MPL', 'CC-BY-2.5']`

And then just filter the list based on that pattern. Depending on the pattern corresponding class would be instantiated:

e.g.

```typescript
let license;
if (LICENSE_TYPE_WHITELIST_PATTERNS.some(pattern => libraryLicenseInfo.license.includes(pattern) ) {
    switch (libraryLicenseInfo.license.includes('GPL')) {
        license = new LGPLLicenseInfos()
    }
}
```

Reply:
ok,
what should happen in this case ?
        "license": "MIT",
        "additionalLicenses": ["GPL-2", "LGPL-2.1"]